lib/msf/core/payload/pingback.rb

```diff
@@ -0,0 +1,51 @@
+# -*- coding => binary -*-
+
+require 'msf/core'
+require 'msf/core/module/platform'
+require 'rex/text'
+
+#
+# This class provides methods for calculating, extracting, and parsing
+# unique ID values used by payloads.
+#
+module Msf::Payload::Pingback
+
+  #
+  # Instance methods
+  #
+  #
+  # Generates a URI with a given checksum and optionally with an embedded UUID if
+  # the desired length can accommodate it.
+  #
+  # @param mode [Symbol] The type of checksum to generate (:connect, :init_native, :init_python, :init_java)
+  # @param len [Integer] The length of the URI not including the leading slash, optionally nil for random
+  # @return [String] A URI with a leading slash that hashes to the checksum, with an optional UUID
+  #
+
+  # Generate a Payload UUID
+  def generate_pingback_uuid
+    self.pingback_uuid ||= SecureRandom.uuid()
+    datastore['PingbackUUID'] = self.pingback_uuid
+    vprint_status("PingbackUUID = #{datastore['PingbackUUID'].gsub('-','')}")
```

Comment:
If we keep doing this `gsub`, should we ever pass around the UUID with the dashes?

Reply:
That's a great question.  I kept it in the original for assuming that since there's a uuid.to_s, there was likely a return method.  I appear to be wrong?

Reply:
Not from that source, I don't think. Unless we need it with the dashes somewhere (I can't find a place), I would do the gsub just once here where we generate it and pass it around without the dashes.